pkg/document/json/rga_tree_split.go

```diff
@@ -580,22 +580,19 @@ func (s *RGATreeSplit) cleanupRemovedNodes(ticket *time.Ticket) int {
 	return count
 }
 
-// purge removes the node passed as a parameter from RGATreeSplit.
+// purge physically purge the given node from RGATreeSplit.
 func (s *RGATreeSplit) purge(node *RGATreeSplitNode) {
-	if node.prev != nil {
-		node.prev.next = node.next
-		if node.next != nil {
-			node.next.prev = node.prev
-		}
-	} else {
-		s.initialHead, node.next.prev = node.next, nil
+	node.prev.next = node.next
+	if node.next != nil {
+		node.next.prev = node.prev
 	}
-	node.next, node.prev = nil, nil
+	node.prev, node.next = nil, nil
 
 	if node.insPrev != nil {
-		node.insPrev.insNext, node.insPrev = nil, nil
+		node.insPrev.insNext = node.insNext
 	}
 	if node.insNext != nil {
-		node.insNext.insPrev, node.insNext = nil, nil
+		node.insNext.insPrev = node.insPrev
```

Comment:
I implemented it in a way that breaks the relationship altogether because the split link represents the relationship that occurs when the node is split. I'm curious about a possible problem when not connecting `insPrev` and `insNext`.

Reply:
`insPrev` and `insNext` are used to remember other nodes connected to the insert. For example, when `abc` is divided into `a`, `b`, `c`, it looks like this: `[abc]` divided to `[a]<->[b]<->[c]` Even if we completely delete `b`, I think it should keep the relationship of `[a]` <-> `[c]`.

`insPrev` is used by `findFloorNodePreferToLeft`. I think that it will not work properly when looking for the left node with priority from the given location.

https://github.com/yorkie-team/yorkie/blob/master/pkg/document/json/rga_tree_split.go#L342

Reply:
Oh! Thank you for the clear explanation :D